docs/spec.md

```diff
@@ -0,0 +1,351 @@
+
+The Ballerina Redis client library supports a multitude of Redis operations related to different data structures, such as key-value operations, list
+operations, and more. The client also supports connecting to Redis clusters, which allows working with multiple Redis nodes using a single
+client. It also comes with the support for various connection features such as connection pooling, which allows reusing connections to the
+Redis server, and SSL/TLS encryption for secure communication with the Redis server.
+
+# 2. Client
+
+The Redis `Client` represents a connection to the Redis server. It maintains a pool of connections throughout its
+lifetime.
+
+## 2.1. Initialize the client
+
+- Redis client can be initialized using the `init` function. Client initialization requires a `ConnectionConfig` object,
+  which contains the connection details for the Redis server.
+
+  ```ballerina
+  # Initialize the Redis client. 
+  #
+  # + config - configuration for the connector
+  # + return - `redis:Error` in case of failures or `nil` if successful. 
+  public isolated function init(*ConnectionConfig config) returns Error? {
+      check self.initClient(self, config);
+  }
+  ```
+
+- Redis connection configurations are defined using the `ConnectionConfig` record, which contains the following fields:
+  ```ballerina
+  # The client endpoint configuration for Redis.
+  #
+  # + connection - connection configuration for the Redis database
+  # + connectionPooling - whether connection pooling is enabled
+  # + isClusterConnection - whether the connection is a cluster connection
+  public type ConnectionConfig record {|
+      ConnectionUri|ConnectionParams connection?;
+      boolean connectionPooling = false;
+      boolean isClusterConnection = false;
+  |};
+  ```
+
+- the connection parameters can be provided either as a single redis URI or, as individual parameters.
+  The `ConnectionUri` and `ConnectionParams` records are defined as follows:
+
+    1. Redis URI based configurations:
+   ```ballerina
+   # The redis Connection URI based configurations. This can become useful when working with
+   # managed Redis databases, where the cloud provider usually provides a connection URI.
+   #
+   # + uri - The connection URI for the Redis database
+   type ConnectionUri record {|
+       string uri;
+   |}; 
+   ```
+
+    2. Redis connection parameters based configurations:
+   ```ballerina
+   # The connection parameters based configurations.
+   #
+   # + host - host address of the Redis database
+   # + port - port of the Redis database
+   # + username - field description
+   # + password - The password for the Redis database
+   # + options - other connection options of the connection configuration
+   type ConnectionParams record {|
+     string host = "localhost";
+     int port = 6379;
+     string username?;
+     string password?;
+     Options options = {};
+   |};
+   ```
+
+  ```ballerina
+  # Connection options for Redis client endpoint.
```

Comment:
We can merge these records into one code-block. Or lets lets divide these record definitions with proper descriptions.

Reply:
Merged into a single code block with https://github.com/ballerina-platform/module-ballerinax-redis/pull/219/commits/5aec5d13ea49f791379343d8759d3c4819f987ad